http/codegen/service_data.go

```diff
@@ -1205,17 +1208,19 @@ func buildPayloadData(e *expr.HTTPEndpointExpr, sd *ServiceData) *PayloadData {
 		name        string
 		ref         string
 	)
-	if payload.Type != expr.Empty {
-		name = svc.Scope.GoFullTypeName(payload, svc.PkgName)
-		ref = svc.Scope.GoFullTypeRef(payload, svc.PkgName)
-	}
-	if init == nil {
-		if o := expr.AsObject(e.Params.Type); o != nil && len(*o) > 0 {
-			returnValue = codegen.Goify((*o)[0].Name, false)
-		} else if o := expr.AsObject(e.Headers.Type); o != nil && len(*o) > 0 {
-			returnValue = codegen.Goify((*o)[0].Name, false)
-		} else if e.MapQueryParams != nil && *e.MapQueryParams == "" {
-			returnValue = mapQueryParam.Name
+	{
+		if payload.Type != expr.Empty {
+			name = payloadAn.Name(true)
```

Comment:
This is an example where I'm not sure the abstraction helps. I'd rather read `svc.Scope.GoFullTypeName` than `payloadAn.Name` as the former is clear while the latter has me go through extra code to understand what's happening.